src/main/kotlin/no/skatteetaten/aurora/cantus/controller/NexusController.kt

```diff
@@ -22,4 +37,79 @@ class NexusController(val nexusService: NexusService) {
         val snapshotVersions = nexusService.getAllVersions(imageGroup, name, SNAPSHOT_REPO)
         return Flux.concat(releaseVersions, snapshotVersions)
     }
+
+    @PostMapping("/image/move")
+    fun moveImage(
+        @Valid @RequestBody moveImageCmd: MoveImageCommand
+        // TODO: add code to verify Flyttebil as source
+    ): Mono<ResponseEntity<MoveImageResult?>> {
+        // Search for image and validate that it correspond with exactly one instance in the expected repo
+        return nexusMoveService.getSingleImage(
+            moveImageCmd.fromRepo,
+            moveImageCmd.name,
+            moveImageCmd.version,
+            moveImageCmd.sha256
+        )
+            .flatMap { singleImageDto ->
+                nexusMoveService.moveImage(
+                    singleImageDto!!.repository,
+                    moveImageCmd.toRepo,
+                    singleImageDto.name,
+                    singleImageDto.version,
+                    singleImageDto.sha256
+                )
+                    .flatMap { movedImageDto ->
+                        logger.info { "Moved image ${movedImageDto.name}:${movedImageDto.version} to ${movedImageDto.repository}" }
+                        Mono.just(ResponseEntity.ok().body(movedImageDto.moveImageResult()))
+                    }
+                    .doOnError {
+                        logger.error { "Failed to move image ${singleImageDto.name}:${singleImageDto.version} with sha ${singleImageDto.sha256}" }
+                        Mono.just(ResponseEntity.internalServerError().body(singleImageDto.moveImageResult()))
```

Comment:
Må man bruke onErrorReturn() for å returnere med denne verdien?

Reply:
Snakket om i standup. Skriver om til å bruke onErrorReturn()